src/components/NavBar/SearchBar.tsx

```diff
@@ -0,0 +1,556 @@
+            {token.price24hChange.toFixed(2)}%
+          </Box>
+        )}
+      </Column>
+    </Row>
+  )
+}
+
+const SkeletonRow = () => {
+  const [isHovered, toggleHovered] = useReducer((s) => !s, false)
+
+  return (
+    <Box className={styles.searchBarDropdown}>
+      <Row
+        as="a"
+        background={isHovered ? 'lightGrayButton' : 'none'}
+        onMouseEnter={toggleHovered}
+        onMouseLeave={toggleHovered}
+        className={styles.suggestionRow}
+      >
+        <Row>
+          <Box className={styles.suggestionImage} style={{ background: '#7C85A24D' }} />
+          <Box borderRadius="round" height="16" width="160" style={{ background: '#7C85A24D' }} />
+        </Row>
+      </Row>
+    </Box>
+  )
+}
+
+interface SearchBarDropdownSectionProps {
+  toggleOpen: () => void
+  suggestions: (GenieCollection | FungibleToken)[]
+  header: string
+  headerIcon?: JSX.Element
+  hoveredIndex: number | undefined
+  startingIndex: number
+  setHoveredIndex: (index: number | undefined) => void
+}
+
+export const SearchBarDropdownSection = ({
+  toggleOpen,
+  suggestions,
+  header,
+  headerIcon = undefined,
+  hoveredIndex,
+  startingIndex,
+  setHoveredIndex,
+}: SearchBarDropdownSectionProps) => {
+  return (
+    <Column gap="12">
+      <Row paddingX="16" paddingY="4" gap="8" color="grey300" className={subheadSmall} style={{ lineHeight: '20px' }}>
+        {headerIcon ? headerIcon : null}
+        <Box>{header}</Box>
+      </Row>
+      <Column gap="12">
+        {suggestions?.map((suggestion, index) =>
+          isCollection(suggestion) ? (
+            <CollectionRow
+              key={suggestion.address}
+              collection={suggestion as GenieCollection}
+              isHovered={hoveredIndex === index + startingIndex}
+              setHoveredIndex={setHoveredIndex}
+              toggleOpen={toggleOpen}
+              index={index + startingIndex}
+            />
+          ) : (
+            <TokenRow
+              key={suggestion.address}
+              token={suggestion as FungibleToken}
+              isHovered={hoveredIndex === index + startingIndex}
+              setHoveredIndex={setHoveredIndex}
+              toggleOpen={toggleOpen}
+              index={index + startingIndex}
+            />
+          )
+        )}
+      </Column>
+    </Column>
+  )
+}
+
+interface SearchBarDropdownProps {
+  toggleOpen: () => void
+  tokens: FungibleToken[]
+  collections: GenieCollection[]
+  hasInput: boolean
+}
+
+export const SearchBarDropdown = ({ toggleOpen, tokens, collections, hasInput }: SearchBarDropdownProps) => {
+  const [hoveredIndex, setHoveredIndex] = useState<number | undefined>(undefined)
+  const searchHistory = useSearchHistory(
+    (state: { history: (FungibleToken | GenieCollection)[] }) => state.history
+  ).slice(0, 2)
+  const isNFTPage = location.hash.includes('/nft')
+  const isTokenPage = location.hash.includes('/token')
+
+  const tokenSearchResults =
```

Comment:
is there a way to move the business logic into a utils file in this folder? hard to follow the UI logic with the algorithm mixed in

Reply:
This case is just a conditional checking if there are search results otherwise show the blank state, I dont think a util function is necessary for this case